src/loaders/lottie/tvgLottieProperty.h

```diff
@@ -826,6 +826,49 @@ struct LottieTextDoc : LottieProperty
 };
 
 
+struct LottieImageData : LottieProperty
+{
+    union {
+        char* b64Data = nullptr;
+        char* path;
+    };
+    char* mimeType = nullptr;
+    uint32_t size = 0;
+    float width = 0.0f;
+    float height = 0.0f;
+
+    ~LottieImageData()
+    {
+        release();
+    }
+
+    void release()
+    {
+        free(b64Data);
+        free(mimeType);
+    }
+
+    uint32_t frameCnt() override { return 0; }
+    uint32_t nearest(float time) override { return 0; }
+    float frameNo(int32_t key) override { return 0; }
+
+    LottieImageData& operator=(const LottieImageData& other)
+    {
+        //shallow copy, used for slot overriding
+        b64Data = other.b64Data;
+        path = other.path;
```

Comment:
One of them (not both) needs to be assigned.

Reply:
added condition:

```cpp
if (other.b64Data) b64Data = other.b64Data;
else path = other.path;
```

Reply:
The union doesn't work as you expected because both variables share the same memory data. b64Data = other.b64Data is sufficient.

Reply:
@tinyjin double check this please.

Reply:
@hermet Ah yes, make sense. I changed condition. Check if mimeType has value and override with embedded data.